src/script/tari_script.rs

```diff
@@ -428,6 +461,55 @@ impl TariScript {
             (..) => Err(ScriptError::IncompatibleTypes),
         }
     }
+
+    fn check_multisig(&self, stack: &mut ExecutionStack, message: Message) -> Result<bool, ScriptError> {
+        use StackItem::*;
+        // get n
+        // must be gt 0
+        let n = match stack.pop().ok_or(ScriptError::StackUnderflow)? {
+            Number(n) if n > 0 => n as usize,
+            Number(_) => return Err(ScriptError::InvalidInput),
+            _ => return Err(ScriptError::IncompatibleTypes),
+        };
+        // pop n pubkeys
+        let public_keys = stack
+            .pop_num_items(n)?
+            .into_iter()
+            .map(|item| match item {
+                PublicKey(pk) => Ok(pk),
+                _ => Err(ScriptError::IncompatibleTypes),
+            })
+            .collect::<Result<Vec<RistrettoPublicKey>, ScriptError>>()?;
+
+        // get m
+        // must be gt 0 and lte n
+        let m = match stack.pop().ok_or(ScriptError::StackUnderflow)? {
+            Number(m) if m > 0 && m <= n as i64 => m as usize,
+            Number(_) => return Err(ScriptError::InvalidInput),
+            _ => return Err(ScriptError::IncompatibleTypes),
+        };
+        // pop m sigs
+        let signatures = stack
+            .pop_num_items(m)?
+            .into_iter()
+            .map(|item| match item {
+                Signature(s) => Ok(s),
+                _ => Err(ScriptError::IncompatibleTypes),
+            })
+            .collect::<Result<Vec<RistrettoSchnorr>, ScriptError>>()?;
+
+        let mut valid_sigs = 0;
+        for signature in signatures.iter() {
```

Comment:
Doesn't this mean that one public key could sign the message n times and it would be valid? We should remove the valid public keys from the list after it's been verified

Reply:
Good catch